aptos-move/aptos-replay-benchmark/src/generator.rs

```diff
@@ -0,0 +1,131 @@
+// Copyright (c) Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{block::Block, overrides::OverrideConfig, workload::Workload};
+use aptos_move_debugger::aptos_debugger::AptosDebugger;
+use aptos_types::transaction::{Transaction, Version};
+use std::{
+    sync::{
+        atomic::{AtomicU64, Ordering},
+        Arc,
+    },
+    time::Instant,
+};
+
+pub struct BenchmarkGenerator {
+    generator: Arc<BenchmarkGeneratorContext>,
+}
+
+impl BenchmarkGenerator {
+    pub fn new(
+        debugger: AptosDebugger,
+        begin_version: Version,
+        end_version: Version,
+        override_config: OverrideConfig,
+    ) -> Self {
+        let generator =
+            BenchmarkGeneratorContext::new(debugger, begin_version, end_version, override_config);
+        Self {
+            generator: Arc::new(generator),
+        }
+    }
+
+    /// Generates a sequence of [Block] for benchmarking.
+    pub async fn generate_blocks(&self) -> anyhow::Result<Vec<Block>> {
+        let limit = self.generator.end_version - self.generator.begin_version + 1;
+        let (txns, _) = self
+            .generator
+            .debugger
+            .get_committed_transactions(self.generator.begin_version, limit)
+            .await?;
+        let txn_blocks = self.generator.partition(txns);
+
+        let num_generated = Arc::new(AtomicU64::new(0));
```

Comment:
hm since it's used within the scope of this function (right?), wouldn't a reference work (instead of Arc)?